src/v/kafka/server/group_manager.cc

```diff
@@ -746,6 +762,75 @@ ss::future<> group_manager::do_recover_group(
     co_return;
 }
 
+ss::future<> group_manager::write_version_fence(
+  model::term_id term, ss::lw_shared_ptr<attached_partition> p) {
+    // how long to delay retrying a fence write if an error occurs
+    constexpr auto fence_write_retry_delay = 10s;
+
+    co_await _feature_table.local().await_feature(
+      features::feature::group_offset_retention, p->as);
+
+    while (true) {
+        if (p->as.abort_requested() || _gate.is_closed()) {
+            break;
+        }
+
+        auto batch = _feature_table.local().encode_version_fence(
+          cluster::cluster_version{9});
```

Comment:
nit: could you add a comment about what's special about cluster version 9? Alternatively maybe we could alias version 9 as some `earliest_version_with_offset_retention` or something